### PR TITLE
Properties conversions

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PRs aims to provide a complete list of inline functions for converting between the material properties including Elastic properties, PoroElastic/ThermoPoroElastic coupling properties: Biot's/Skempton's parameters, drained/undrained, etc. These functions are needed for the constitutive models.

### CODE REVIEW

**Comment by joshua-white** [comment]
Hey @sytuannguyen, there is a lot of good material here.  I noticed you changed the naming convention when we get to your additions.  For the first part, we write:
```cpp
real64 K = conversions::YoungModAndPoissonRatio::toBulkMod( Y, P )
```
where in your additions we write
```cpp
real64 B = conversions::BiotCoefficient::useKKs( K, Ks )
```
Can we do two things?:

- Make the everything stylistically similar
- Use more obvious variable naming

I'd suggest something like
```cpp
real64 B = conversions::BulkModAndUndrainedBulkMod::toBiotCoefficient( K, Ks )
```
The downside is the length (particularly with three arguments) but it remains quickly understandable.

**Comment by sytuannguyen** [comment]
@joshua-white yes it will take a line long name when we go to THM coupling ^_^. Also I realized that considering a namespace for an input combination is appropriate for elastic but not for coupling problems. I seems better to consider a namespace for each property as I tried here. 

Or maybe we can consider each property as an object and use class/template to get rid of the name complexity? We should optimize for the most complex case first then homogenize back for elastic case.

**Review by TotoGaz — COMMENTED** [review]
Do you really need to set a full `builder pattern` with loads of side effects, that needs to be understood by the user and can be wrongly used (forgetting to set a value is now possible, I do not think it was before)?

Which feature is added by your code that was not there with standard functions? What was the problem with them?

Plus you have to create a 1300 lines long for such a simple task, that really looks overkill to me.

**Comment by sytuannguyen** [comment]
> Which feature is added by your code that was not there with standard functions? What was the problem with them?

Originally @joshua-white considered namespace like BulkModAndShearMod, YoungModAndShearMod, etc. for this conversion. This naming style works perfectly for elastic properties. However it becomes very complicated when I tried to use it for Thermo-Hydro-Mechanical coupling properties. See discussions above for more details.

 I come to this solution to avoid the naming complexity but I agree that it could be optimized. @TotoGaz please suggest a specific solution if you have any.

File: src/coreComponents/constitutive/solid/ElasticIsotropic.cpp [hunk 1]
```diff
@@ -106,33 +106,66 @@ void ElasticIsotropic::postProcessInput()
...
+    G = ShearModulus().
+          setBulkModulus( K ).
+          setPoissonRatio( nu ).
+          getValue();
   }
   else if( E >= 0.0 && K >=0 )
   {
-    nu = conversions::BulkModAndYoungsMod::toPoissonRatio( K, E );
-    G  = conversions::BulkModAndYoungsMod::toShearMod( K, E );
+    nu = PoissonRatio( BulkModulus( K ), YoungModulus( E ) ).value;
```
[the remark this answers is not in the document]
Comment (TotoGaz) [reply src/coreComponents/constitutive/solid/ElasticIsotropic.cpp:145]: My point of view is that the `unit` question is far bigger than just this bulk and Young modulus. A discussion is planned and we should wait for the decision, and possibilities are large.

For example, do we want different types (`BulkModulus`, `YoungModulus `) for quantities with the same unit (pressure)? This is not obvious. Do we want to type our fields, use non-dimensional ones (my preference)?
What about non-dimensional quantities like `PoissonRation`: do you want a type, use raw `real64`?
Do we want to deal with unit questions too (`GPa`, `MPa`), probably, but how will your code evolve?

As such you're `unit` solution will quite surely require revamping.
I would personally don't bother right now and just use `real64` as entry of free function.

But could it be a nice and fresh first place to test once we've decided what we want?

**Pull request closed without merging**